### DISCUSSION TITLE
[HELP]: Deleted Bull Jobs Are Still Executing with Empty Data

### DISCUSSION THREAD

**Discussion by thenithishrajan** [taskforcesh/bullmq#2986] [Q&A]
I'm using a custom function to delete Bull jobs in bulk because I need to remove a large number of jobs from different queues. However, I've encountered an issue where deleted jobs are still being executed at their scheduled delayed time, but with empty job data.

**Custom Function:**

```javascript
deleteBullJobsByIds: async (jobIds: string[]) => {
  try {
    for (let i = 0; i < jobIds.length; i += 50) {
      const pipeline = redisClient.multi();
      const slice = jobIds.slice(i, i + 50);

      slice.forEach((id) => pipeline.del(id));
      await pipeline.exec();
    }
  } catch (error) {
    throw new Error(`Failed to delete Bull jobs: ${error}`);
  }
},
```

**Problem:** When I delete a job (e.g., `bull:workflowqueue:555`) using the above code, it is successfully removed from Redis. However, when the job's delay time (e.g., 15000 ms) elapses, the job is still executed, but with empty data instead of the original job data.

Any guidance or suggestions would be greatly appreciated!

**Comment by manast** [comment]
You should not write your own scripts to remove jobs as thats almost guaranteed going to lead to issues like the one you have. If you still want to do it I recommend you to check the source code to see how BullMQ removes jobs.
